modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/dag/decoders.scala

```diff
@@ -0,0 +1,231 @@
+package org.tessellation.rosetta.server.model.dag
+
+import org.tessellation.rosetta.server.model._
+import org.tessellation.rosetta.server.model.dag.metadataSchema._
+import org.tessellation.rosetta.server.model.dag.schema._
+
+/** This whole thing has to be manually re-arranged after generating because of implicit dependencies requiring to be in order.
+  */
```

Comment:
I doubt we want to do this manually. I would rather see a complete yaml specification of OpenAPI as well as the generator configured either via the [SBT plugin](https://github.com/OpenAPITools/sbt-openapi-generator) or via the custom sbt task, that will use a custom Mustache [template](https://openapi-generator.tech/docs/templating/) that will add all the necessary derivations to the generated classes (so they don't need to be stored as .scala sources in the repo but generated during the build).

Reply:
See the above comment on `sbt autogen` being broken. An automated way to do this is necessary, but I think it will take considerable effort and should be considered out of scope for this PR.

Reply:
@cngo-github what's exactly broken with `sbt autogen`?
code like this (the file above) is risky and hard to maintain. even a custom sbt task would way better, but ideally it should be done via templates as @tbekas suggested.
if you want to do it out of scope for this PR, then please create separate PR for auto generated classes first and then separate one to regenerate all already created classes.

Reply:
@ryle-goehausen , can you please elucidate what about `sbt autogen`?

Reply:
Sure, the list of generators I tried was from here under the Scala section: https://github.com/OpenAPITools/openapi-generator/tree/master/docs/generators -- I believe I went through like 7 different generators I forget all the names, but it was from that repo. I also tried some of the code gen builtins in some IDEs, each one had a number of flaws -- either generating unnecessary or redundant code or throwing exceptions halfway through or failing to generate the correct classes. The last time I looked into this there didn't seem to be an easy solution, and a bunch of complaints for other OpenAPI failures related to Scala. I think it's specific to the yaml file that's being used, as it works in plenty of other cases but didn't for this. Possible I misconfigured it -- but I think it's appropriate to take a bit of a shortcut here rather than dealing with patching / custom SBT configurations since we don't anticipate needing to update for a while. 

I think it'd be reasonable to revisit this assuming there's a requirement to upgrade rosetta classes in the future, in which case what is present should be reasonably close to an sbt build step.

Reply:
@ryle-goehausen this is exactly what I addressed in my comment. Just use a custom Mustache [template](https://openapi-generator.tech/docs/templating/). If an sbt-plugin is "broken" as @cngo-github says we can write a custom sbt task as @marcinwadon suggested, but that's only when sbt-plugin can't be used.

Making everything manual makes `rosetta` module very difficult to maintain and prone to introducing errors (desyncing classes with their autogenerated counterparts).

Reply:
I will need assistance with this. We can go over it when we meet @tbekas next week. Additionally, please point me to a good tutorial.

Reply:
The generation of the `Rosetta` classes has ben made a custom `sbt` task. This is no longer needed.